Solution1/ConsoleApp/Services/UserCommunicateService.cs

```diff
@@ -13,60 +21,111 @@ public class UserCommunicateService : IUserCommunicateService
     {
         private readonly ILogger _logger;
         private readonly IWeatherServiсe _weatherServiсe;
+        private readonly IValidator<DataForWeatherRequestDTO> _validator;
+        private readonly Invoker _invoker;
 
-        public UserCommunicateService(ILogger logger, IWeatherServiсe weatherService)
+
+        public UserCommunicateService(ILogger logger, IWeatherServiсe weatherService, IValidator<DataForWeatherRequestDTO> validator)
         {
             _logger = logger;
             _weatherServiсe = weatherService;
+            _validator = validator;
+            _invoker = new Invoker();
         }
 
-        public async Task<bool> CommunicateAsync()
+        public async Task ShowMenuAsync()
```

Comment:
Show menu does not only show, menu it does the whole app logic

Reply:
+